src/node_crypto.cc

```diff
@@ -608,8 +584,74 @@ int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
   }
 
  end:
-  if (x != nullptr)
+  if (ret && x != nullptr) {
     *cert = x;
+    CRYPTO_add(&(*cert)->references, 1, CRYPTO_LOCK_X509);
```

Comment:
The fact that the OpenSSL API lacks this doesn't mean that no-one will ever want to use it. We also use this for `issuer` for quite a long time. How do you suggest to hold a reference to it, otherwise?